python-client/tests/communications/test_dto_parsing.py

```diff
@@ -0,0 +1,414 @@
+import pytest
+
+from pydantic import ValidationError
+
+from giskard.ml_worker import websocket
+
+TEST_PROJECT_KEY = "123"
+TEST_MODEL_ID = "213"
+TEST_DATASET_ID = "321"
+
+
+@pytest.mark.parametrize(
+    "artifact_ref_dict",
+    [
+        {},
+        {"project_key": TEST_PROJECT_KEY, "sample": True},  # Missing id
+    ],
+)
+def test_artifact_ref_missing_fields(artifact_ref_dict):
+    with pytest.raises(ValidationError):
+        websocket.ArtifactRef.parse_obj(artifact_ref_dict)
+
+
+@pytest.mark.parametrize(
+    "artifact_ref_dict",
+    [
+        {"project_key": TEST_PROJECT_KEY, "id": TEST_MODEL_ID},  # Missing sample
+        {"id": TEST_MODEL_ID, "sample": True},  # Missing project_key
+        {"project_key": TEST_PROJECT_KEY, "id": TEST_MODEL_ID, "sample": True},
+    ],
+)
+def test_artifact_ref(artifact_ref_dict):
+    artifact_ref = websocket.ArtifactRef.parse_obj(artifact_ref_dict)
+    assert isinstance(artifact_ref, websocket.ArtifactRef)
+    assert artifact_ref.project_key is None or artifact_ref.project_key == TEST_PROJECT_KEY
+    assert artifact_ref.id == TEST_MODEL_ID
+    assert artifact_ref.sample is None or artifact_ref.sample is True or artifact_ref.sample is False
+
+
+TEST_FUNC_ARGUMENT_NAME = "arg"
+TEST_FUNC_ARGUMENTS_NAME1 = "arg0"
+TEST_FUNC_ARGUMENTS_NAME2 = "arg1"
+TEST_FUNC_ARGUMENT_FLOAT = 114.514
+TEST_FUNC_ARGUMENT_INT = 114
+TEST_FUNC_ARGUMENT_STR = "giskard"
+TEST_FUNC_ARGUMENT_BOOL = False
+TEST_FUNC_ARGUMENT_SLICING_FUNCTION_ID = "114"
+TEST_FUNC_ARGUMENT_TRANSFORMATION_FUNCTION_ID = "514"
+
+
+@pytest.mark.parametrize(
+    "func_argument_dict",
+    [
+        {},
+        {"none": True},  # Missing name
+        {"name": TEST_FUNC_ARGUMENT_NAME},  # Missing none
+    ],
+)
+def test_func_argument_missing_fields(func_argument_dict):
+    with pytest.raises(ValidationError):
+        websocket.FuncArgument.parse_obj(func_argument_dict)
+
+
+@pytest.mark.parametrize(
+    "func_argument_dict",
+    [
+        {"name": TEST_FUNC_ARGUMENT_NAME, "none": True},  # None argument
```

Comment:
If you want to name the param, you can do
```python
@pytest.param({"name": TEST_FUNC_ARGUMENT_NAME, "none": True}, id="None")
```

It will appear as `test_func_argument[None]` in the test log